Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -477,6 +477,194 @@ PTF_TEST_CASE(TestPcapLiveDeviceBlockingMode)
 } // TestPcapLiveDeviceBlockingMode
 
 
+PTF_TEST_CASE(TestPcapLiveDeviceWithLambda)
+{
+	pcpp::PcapLiveDevice* liveDev = nullptr;
+	pcpp::IPv4Address ipToSearch(PcapTestGlobalArgs.ipToSendReceivePackets.c_str());
+	liveDev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByIp(ipToSearch);
+	PTF_ASSERT_NOT_NULL(liveDev);
+	PTF_ASSERT_GREATER_THAN(liveDev->getMtu(), 0);
+	PTF_ASSERT_TRUE(liveDev->open());
+	DeviceTeardown devTeardown(liveDev);
+	int packetCount = 0;
+	int numOfTimeStatsWereInvoked = 0;
+
+	auto packetArrivesLambda = [](pcpp::RawPacket* rawPacket, pcpp::PcapLiveDevice* pDevice, void* userCookie)
+	{
+		(*(int*)userCookie)++;
+	};
+
+	auto statsUpdateLambda = [](pcpp::IPcapDevice::PcapStats& stats, void* userCookie)
+	{
+		(*(int*)userCookie)++;
+	};
+
+	PTF_ASSERT_TRUE(liveDev->startCapture(packetArrivesLambda , (void*)&packetCount, 1, statsUpdateLambda, (void*)&numOfTimeStatsWereInvoked));
+	int totalSleepTime = 0;
+	while (totalSleepTime <= 20)
+	{
+		pcpp::multiPlatformSleep(2);
+		totalSleepTime += 2;
+		if (packetCount > 0)
+			break;
+	}
+
+	PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
+
+	liveDev->stopCapture();
+	PTF_ASSERT_GREATER_THAN(packetCount, 0);
+	PTF_ASSERT_GREATER_OR_EQUAL_THAN(numOfTimeStatsWereInvoked, totalSleepTime-2);
+} // TestPcapLiveDeviceWithLambda
+
+
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModeWithLambda)
```

Comment:
I don't think we should duplicate the entire test to test lambda. We just need to add one lambda test to the existing [`TestPcapLiveDeviceBlockingMode`](https://github.com/seladb/PcapPlusPlus/blob/dd9b3d8aa1b7403d08be7960bf01cc1be94575cd/Tests/Pcap%2B%2BTest/Tests/LiveDeviceTests.cpp#L379) as I suggested [here](https://github.com/seladb/PcapPlusPlus/pull/1273#discussion_r1436690333):

```cpp
PTF_TEST_CASE(TestPcapLiveDeviceBlockingMode)
{
	...

	auto packetArrivesBlockingModeLambda = [](pcpp::RawPacket* rawPacket, pcpp::PcapLiveDevice* dev, void* userCookie)
	{
		int* packetCount = (int*)userCookie;
		if ((*packetCount) == 5)
			return true;

		(*packetCount)++;
		return false;
	};

	int packetCount = 0;
		
	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeLambda, &packetCount, 30), 1);
	PTF_ASSERT_EQUAL(packetCount, 5);

	...
} // TestPcapLiveDeviceBlockingMode
```

Reply:
I misunderstood. okay, let me just keep one `startCaptureBlockingMode` call in `TestPcapLiveDeviceBlockingMode`